cadc-inventory/src/main/java/org/opencadc/inventory/Entity.java

```diff
@@ -91,13 +91,22 @@
 public class Entity {
     private static final Logger log = Logger.getLogger(Entity.class);
 
-    private static final boolean MCS_DEBUG = true;
+    private static final boolean MCS_DEBUG = false;
     
     private UUID id;
     private Date lastModified;
     private URI metaChecksum;
     
-    protected Entity() { 
+    protected Entity() {
+        this(null);
+    }
+    
+    protected Entity(UUID id) {
+        if (id == null) {
+            this.id = UUID.randomUUID();
+        } else {
+            this.id = id;
+        }
     }
```

Comment:
Maybe just make the no-arg contructor public?  Then you could check for accidentally null UUIDs in the constructor with a parameter.

Reply:
simplified constructors; they are protected because technically you don't instantiate a bare Entity... making it abstract